web/src/main/java/org/akaza/openclinica/service/ValidateServiceImpl.java

```diff
@@ -251,7 +251,61 @@ public void validateStudyAndRoles(String studyOid, String siteOid, UserAccountBe
         }
 
     }
+    
+    /**
+     *  this method is used when get/extract participant information
+     *  
+     * @param studyOid
+     * @param siteOid
+     * @param userAccountBean
+     * @param includeAccessCode
+     */
+    public void validateStudyAndRoles(String studyOid, String siteOid, UserAccountBean userAccountBean,boolean includeAccessCode) {
```

Comment:
this is not just for access code, but for both contact info and access code?
maybe call it includePII?
Also, this method should only be used for reads and not writes, so call it something like validateStudyAndRolesForRead

Reply:
FYI
finished code change based on code review and more testing.